mantidproject/mantid
# Sample Transmission Calculator no longer accepts negative wavelength values

**Description of work.**
When the input type is set to Multiple, the interface no longer executes if negative wavelengths have been inputted.

**To test:**
1. Interfaces -> General -> Sample Transmission Calculator
2. Under Input Wavelength Range, set Type to Multiple.
3. Enter a comma-separated list of three values (min wavelength, width, max wavelength), making either or both wavelengths negative. E.g. -10,0.1,-5.
4. Enter H for Chemical Formula and click Calculate.
5. Check an error is thrown, and check that the interface works correctly when negative wavelengths aren't used.


Fixes #24224 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
